### PR TITLE
multi: split commits from loop repo

### PR DESCRIPTION
**Pull request by joostjager**
Pure code extraction from loop repo using `git subtree split`

Loop counterpart: https://github.com/lightninglabs/loop/pull/237

### CODE REVIEW

File: chainnotifier_client.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
+package lndclient
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/loop/swap"
```
Comment (guggero) [inline chainnotifier_client.go:10]: Can this import be removed? Or do you want to do this in a follow-up PR so this is a pure code move?

**Comment by joostjager** [comment]
I don't know if the tagging is too much, because every merge will probably need a new tag?

**Comment by guggero** [comment]
@carlaKC: The version is weird because the commit you're referencing to isn't in the main repo yet. You have to add a rewrite in faraday's `go.mod` to use Joost's fork.

**Comment by carlaKC** [comment]
> I don't know if the tagging is too much, because every merge will probably need a new tag?

We don't need to tag each merge, because go-mod can do pseudo versions. I think it might be helpful to tag the point where we split out from loop (maybe not) so that other projects using lndclient can just point to that version?

**Comment by joostjager** [comment]
But after the first change to this repo, the need to switch to pseudo version anyway? Btw, after merge it looks like `github.com/lightninglabs/lndclient v0.0.0-20200618064231-15628713b198`

I am fine with tagging (which version to start with?), but don't think it is that useful.

**Comment by joostjager** [comment]
Pushed tag `v1.0.0`.

**Pull request merged**